app/controllers/search_controller.rb

```diff
@@ -2,11 +2,30 @@ class SearchController < ApplicationController
   def index
     return if params[:query].nil?
 
-    @results = Person.search(params[:query])
-    @results += Room.search(params[:query])
-    @results += Chair.search(params[:query])
+    @exact_results = add_results_for(params[:query])
+
+    words_in_query = params[:query].scan(/[A-Za-z0-9]+/)
+
+    @more_results = words_in_query.flat_map { |word| add_results_for(word) }
+    @more_results = sort_by_frequency(@more_results) - @exact_results
+
     @params = params[:query]
   end
 
   helper_method :index
+
+  private
+
+  def add_results_for(query)
+    results = Person.search(query)
+    results += Room.search(query)
+    results += Chair.search(query)
+    results
```

Comment:
just a suggestion: Use this format instead, so no need to results
Person.search(query) +
  Room.search(query) +
  Chair.search(query)